src/posix/platform/infra_if.cpp

```diff
@@ -101,16 +101,12 @@ otError otPlatInfraIfSendIcmp6Nd(uint32_t            aInfraIfIndex,
 }
 #endif
 
+#if OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE && OPENTHREAD_POSIX_CONFIG_NAT64_AIL_PREFIX_ENABLE
 otError otPlatInfraIfDiscoverNat64Prefix(uint32_t aInfraIfIndex)
```

Comment:
Regarding the use-case, I guess the idea is when `OPENTHREAD_POSIX_CONFIG_NAT64_AIL_PREFIX_ENABLE ` is disabled, then  the platform related APIs for NAT64 should be provided by another library? Is this correct?
Would the new platform implemenation take some action on `otPlatInfraIfDiscoverNat64Prefix()` or just provide an empty function?

Reply:
> The platform related APIs for NAT64 should be provided by another library? 
Yes. The implementation will look like this:
```c++
extern "C" otError otPlatInfraIfDiscoverNat64Prefix(uint32_t aInfraIfIndex)
{
    OtDaemonServer *otDaemonServer = OtDaemonServer::Get();
    otError         error          = OT_ERROR_NONE;

    VerifyOrExit(otDaemonServer != nullptr, error = OT_ERROR_INVALID_STATE);

     otPlatInfraIfDiscoverNat64PrefixDone(GetOtInstance(), infraIfIndex, &mNat64Prefix);
exit:
    return error;
}
```

It's not an empty implementation but it needs to call `otPlatInfraIfDiscoverNat64PrefixDone` to notify the OT core NAT64 module to re-evaluate its state. That is necessary when `Nat64PrefixManager::Start()` is triggered. Otherwise, if the `otPlatInfraIfDiscoverNat64Prefix`'s implementation is empty, the re-evaluation won't be triggered.

The `mNat64Prefix` is a cache of the NAT64 prefix maintained by the `ot-br-posix`'s [vendor server](https://cs.android.com/android/platform/superproject/main/+/main:external/ot-br-posix/src/android/otdaemon_server.cpp;l=2?q=otdaemon_server.cpp). When the platform discovers a NAT64 prefix, platform will notify the vendor server to update the `mNat64Prefix` and vendor server will notify OT core by `otPlatInfraIfDiscoverNat64PrefixDone`.